statistics/ma_performance_calculator.py

```diff
@@ -0,0 +1,118 @@
+        self.logger.debug("Starting to calculate the semantic preconditions performance")
+        for observation in self.dataset_observations:
+            possible_ground_actions = self._create_grounded_action_vocabulary(self.model_domain, observation.grounded_objects)
+            observation_objects = observation.grounded_objects
+            for component in observation.components:
+                possible_ground_actions.extend(component.grounded_joint_action.actions)
+                self.logger.info(
+                    f"Calculating the preconditions' ma semantic performance for the action the state - {component.previous_state.serialize()}"
+                )
+                for action in possible_ground_actions:
+                    if action.name not in learned_domain.actions:
+                        continue
+
+                    true_positive, false_positive, false_negative = self._calculate_action_applicability_rate(
+                        action, learned_domain_path, component.previous_state, observation_objects,
+                    )
+                    num_true_positives[action.name] += true_positive
+                    num_false_positives[action.name] += false_positive
+                    num_false_negatives[action.name] += false_negative
+
+        return _calculate_precision_recall(num_false_negatives, num_false_positives, num_true_positives)
+
+    def calculate_performance(self, learned_domain_path: Path, num_used_observations: int) -> None:
+        """Calculates the model's performance with both the precision and the recall values calculated.
+
+        :param learned_domain_path: the path to the learned action model.
+        :param num_used_observations: the number of observations used to learn the action model.
+        """
+        learned_domain = DomainParser(domain_path=learned_domain_path, partial_parsing=False).parse_domain()
+        self.logger.info("Starting to calculate the ma semantic preconditions performance of the learned domain.")
+        preconditions_precision, preconditions_recall = self.calculate_preconditions_semantic_performance(learned_domain, learned_domain_path)
+        self.logger.info("Starting to calculate the ma semantic effects performance of the learned domain.")
+        for action_name in self.model_domain.actions:
+            action_stats = {
+                "learning_algorithm": self.learning_algorithm.name,
+                "action_name": action_name,
+                "num_trajectories": num_used_observations,
+                "precondition_precision": preconditions_precision[action_name],
+                "precondition_recall": preconditions_recall[action_name],
+            }
+            self.combined_stats.append(action_stats)
+
+        self.logger.info(f"Finished calculating the ma learning performance for {self.learning_algorithm.name}.")
+
+    def export_semantic_performance(self, fold_num: int, iteration_num: int = 0) -> None:
```

Comment:
you can just use the parent method, set some value to the effects (something like TBD or something), that way you don't have to re-implement the same method.